servicex/models.py

```diff
@@ -58,6 +60,20 @@ def delete_from_db(self):
         db.session.delete(self)
         db.session.commit()
 
+    def send_email(self, template_path):
+        mailgun_api_key = current_app.config.get('MAILGUN_API_KEY')
+        mailgun_domain = current_app.config.get('MAILGUN_DOMAIN')
+        if not mailgun_api_key or not mailgun_domain:
+            return
+        res = requests.post(
```

Comment:
Ok, I really hate seeing raw `request` calls in the DB model. I know that in the absence of a workable dependency injection framework, it's cumbersome to inject yet another  adaptor. Perhaps just make a MailGunAdapter and instantiate one here for this one use case.